### PR TITLE
Simplify some ternary operations using elvis operator and remove unnecessary parentheses in libraries/legacy

### PR DESCRIPTION
**Pull request by frankmayer**
### Summary of Changes
- Simplify some ternary operations using elvis operator and remove unnecessary parentheses in libraries/legacy

This PR is part of a set to try to separate some of the changes done in one of my previous batch PR's for the libraries/legacy directory, which is still on hold (#12220).
Once the new set is merged it will hopefully reduce the changes in that PR, so it can be reviewed easier and finally be merged.

The changes in this PR should be fairly easy to review. In hope that  this will get merged quickly. ;)

### Testing Instructions

None, should not change behavior

### Documentation Changes Required

None.

### CODE REVIEW

**Review by andrepereiradasilva — APPROVED** [review]

**Comment by frankmayer** [comment]
Conflicts resolved. Pls check and merge.

File: libraries/legacy/categories/categories.php [hunk 1]
```diff
@@ -103,10 +103,10 @@ public function __construct($options)
 		$this->_table = $options['table'];
 		$this->_field = (isset($options['field']) && $options['field']) ? $options['field'] : 'catid';
 		$this->_key = (isset($options['key']) && $options['key']) ? $options['key'] : 'id';
```
Comment (Quy) [inline libraries/legacy/categories/categories.php]: Lines 104 and 105 can be simplified.

File: libraries/legacy/model/list.php [hunk 2]
```diff
@@ -694,7 +694,7 @@ public function getUserStateFromRequest($key, $request, $default = null, $type =
 			}
 		}
 
-		if (($cur_state != $new_state) && $new_state !== null && ($resetPage))
+		if (($cur_state != $new_state) && $new_state !== null && $resetPage)
```
Comment (Quy) [inline libraries/legacy/model/list.php]: Remove parentheses `($cur_state != $new_state)`.

File: libraries/legacy/categories/categories.php [hunk 3]
```diff
@@ -103,10 +103,10 @@ public function __construct($options)
...
-		$this->_statefield = (isset($options['statefield'])) ? $options['statefield'] : 'state';
-		$options['access'] = (isset($options['access'])) ? $options['access'] : 'true';
-		$options['published'] = (isset($options['published'])) ? $options['published'] : 1;
-		$options['countItems'] = (isset($options['countItems'])) ? $options['countItems'] : 0;
+		$this->_statefield = isset($options['statefield']) ? $options['statefield'] : 'state';
+		$options['access'] = isset($options['access']) ? $options['access'] : 'true';
+		$options['published'] = isset($options['published']) ? $options['published'] : 1;
+		$options['countItems'] = isset($options['countItems']) ? $options['countItems'] : 0;
 		$options['currentlang'] = JLanguageMultilang::isEnabled() ? JFactory::getLanguage()->getTag() : 0;
```
Comment (Quy) [inline libraries/legacy/categories/categories.php]: Apply elvis operator on lines 106-109.

File: libraries/legacy/categories/categories.php (hunk 1 above)
Comment (frankmayer) [inline libraries/legacy/categories/categories.php]: If I am not mistaken, in these cases, you can't just simplify it. Since you need the isset() to make sure the value is set... And if simplifying the ternary with the result of `(isset($options['field']) && $options['field'])` you get a `1` in case of true, not the value of `$options['field']`

Comment (Quy) [inline libraries/legacy/categories/categories.php]: Like this: `$this->_field = isset($options['field']) && $options['field'] ? $options['field'] : 'catid';`

Comment (frankmayer) [inline libraries/legacy/categories/categories.php]: Ah, you meant removal of parentheses... :smile: I was reading it like simplifying the ternary operation...

Comment (Quy) [inline libraries/legacy/categories/categories.php]: Sorry. I updated the comment after my initial posting.

Comment (frankmayer) [inline libraries/legacy/categories/categories.php]: No worries...

Comment (andrepereiradasilva) [inline libraries/legacy/categories/categories.php]: will you remove the parenthesis then?

Comment (frankmayer) [inline libraries/legacy/categories/categories.php]: @andrepereiradasilva LOL, yes...

File: libraries/legacy/model/legacy.php [hunk 4]
```diff
@@ -578,8 +578,8 @@ protected function cleanCache($group = null, $client_id = 0)
 		$conf = JFactory::getConfig();
 
 		$options = array(
-			'defaultgroup' => ($group) ? $group : (isset($this->option) ? $this->option : JFactory::getApplication()->input->get('option')),
-			'cachebase' => ($client_id) ? JPATH_ADMINISTRATOR . '/cache' : $conf->get('cache_path', JPATH_SITE . '/cache'),
+			'defaultgroup' => $group ?: (isset($this->option) ? $this->option : JFactory::getApplication()->input->get('option')),
```
Comment (Quy) [inline libraries/legacy/model/legacy.php]: Align `=>`. (Sorry I missed these before :frowning: )

File: libraries/legacy/table/menu/type.php [hunk 5]
```diff
@@ -167,7 +167,7 @@ public function store($updateNulls = false)
 	public function delete($pk = null)
 	{
 		$k = $this->_tbl_key;
-		$pk = (is_null($pk)) ? $this->$k : $pk;
```
Comment (Quy) [inline libraries/legacy/table/menu/type.php]: Align `=`

File: libraries/legacy/model/legacy.php (hunk 4 above)
Comment (frankmayer) [inline libraries/legacy/model/legacy.php]: I think  there are many such cases in lots of files, but lets do those in another PR...
I'll do the ones you have mentioned already, but lets try to get this RTC ;)

File: libraries/legacy/table/menu/type.php (hunk 5 above)
Comment (frankmayer) [inline libraries/legacy/table/menu/type.php]: as above

File: libraries/legacy/model/legacy.php (hunk 4 above)
Comment (frankmayer) [inline libraries/legacy/model/legacy.php]: I think that there might already be another PR for those cases. If not, we can do that in new ones.

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 99ec6fcc956699e2e49e995155dda9b2d9d554d9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13240">issues.joomla.org/tracker/joomla-cms/13240</a>.</sub>

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on 99ec6fcc956699e2e49e995155dda9b2d9d554d9<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13240">issues.joomla.org/tracker/joomla-cms/13240</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
RTC after two successful tests.
